2-mandatory/3-shopping-list.js

```diff
@@ -19,6 +19,20 @@ let pantry = {
 };
 
 function createShoppingList(recipe) {
+  let missimgItemArray = [];
+
+  for (let ingredient of recipe.ingredients) {
+    if (
+      !pantry.fridgeContents.includes(ingredient) &&
+      !pantry.cupboardContents.includes(ingredient)
+    ) {
+      missimgItemArray.push(ingredient);
+    }
+  }
+
+  delete recipe.ingredients;
+  recipe.items = missimgItemArray;
+  return recipe;
```

Comment:
your for loop here is really good, however instead of mutating the passed in recipe object, you should be returning a new object, for example:
```
 return {
    name: recipe.name,
    items: missingItemArray
  }
```
with your above approach you are actually changing the original recipe which is passed into your function